iot-hub/Samples/service/ServiceClientSample/ServiceClientSample.cs

```diff
@@ -50,13 +54,50 @@ public async Task RunSampleAsync(TimeSpan runningTime)
 
         }
 
+        private async Task ReceiveMessageFeedbacksAsync(CancellationToken token)
+        {
+            _logger.LogInformation("Starting to listen to feedback messages");
+
+            while (!token.IsCancellationRequested)
+            {
+                var rec = _serviceClient.GetFeedbackReceiver();
+
+                try
+                {
+                    FeedbackBatch messages = await rec.ReceiveAsync();
+                    if (messages != null)
+                    {
+                        _logger.LogInformation("New Feedback received:");
+                        _logger.LogInformation($"\tEnqueue Time: {messages.EnqueuedTime}");
+                        _logger.LogInformation($"\tNumber of messages in the batch: {messages.Records.Count()}");
+                        foreach (FeedbackRecord record in messages.Records)
+                        {
+                            _logger.LogInformation($"\tDevice {record.DeviceId} acted on message: {record.OriginalMessageId} with status: {record.StatusCode}");
+                        }
+
+                        await rec.CompleteAsync(messages);
+                    }
+                }
+                catch (Exception e)
+                {
+                    _logger.LogError($"Unexpected error, will need to reinitialize the client: {e}");
+                    await InitializeServiceClientAsync();
+                }
+            }
+        }
+
         private async Task SendC2dMessagesAsync(CancellationToken cancellationToken)
         {
             int messageCount = 0;
             while (!cancellationToken.IsCancellationRequested)
             {
                 var str = $"Hello, Cloud! - Message {++messageCount }";
-                var message = new Message(Encoding.ASCII.GetBytes(str));
+                var message = new Message(Encoding.ASCII.GetBytes(str))
+                {
+                    // An acknowledgment is sent on delivery success or failure.

```

Comment:
I did not know that there are two correct spellings for Acknowledgment (Acknowledgment and Acknowledgment).
The version with no E after G is what my spelling tool recognizes as the correct one but both exist in the dictionary.

Reply:
Without the 'e' in NA.

https://blog.inkforall.com/acknowledgement-or-acknowledgment

Reply:
We use the non-NA version (with an E) in our codebase.